xlators/features/marker/src/marker.c

```diff
@@ -2963,24 +2963,31 @@ marker_lookup_cbk(call_frame_t *frame, void *cookie, xlator_t *this,
     return 0;
 }
 
-int32_t
+static int32_t
 marker_lookup(call_frame_t *frame, xlator_t *this, loc_t *loc,
               dict_t *xattr_req)
 {
     int32_t ret = 0;
     marker_local_t *local = NULL;
     marker_conf_t *priv = NULL;
+    gf_boolean_t unref = _gf_false;  // do we need to unref the dict
 
     priv = this->private;
 
+    if (priv->version <= 0)
+        goto check_feature;
```

Comment:
If we skip `xattr_req` initialization, it may be `NULL`, which is incorrect if we later call `mq_req_xattr()`.

Reply:
I thought it can't happen, since we check first with:
```
    if (priv->feature_enabled == 0)
        goto wind;
```

So won't be calling mq_req_xattr()

Reply:
If `feature_enabled` can't be != 0 when `version` is <= 0 (or it can be ignored), why not just jump to `wind` from the first check at line 2978 ?

I've done a quick check of the code and it doesn't seem that `feature_enabled` couldn't be set when `private` is <= 0. In some cases an error log is written if this happens, but the code continues without throwing an error, allowing that combination to exist (in one case it even doesn't show an error).

I think in this case we should just jump to `wind` at line 2978.

Reply:
Done.